bolt-taproot-gossip.md

```diff
@@ -0,0 +1,1083 @@
+* `utf8`: a byte as part of a UTF-8 string. A writer MUST ensure an array of
+  these is a valid UTF-8 string, a reader MAY reject any messages containing an
+  array of these which is not a valid UTF-8 string.
+
+## TLV Based Messages
+
+The initial set of Lightning Network messages consisted of fields that had to 
+be present. Later on, TLV encoding was introduced which provided a backward 
+compatible way to extend messages. To ensure that the new messages defined in 
+this document remain as future-proof as possible, the messages will be pure TLV
+streams. By making all fields in the messages TLV records, fields that we
+consider mandatory today can easily be dropped in future (when coupled with a 
+new feature bit) without needing to completely redefine the gossip message in 
+order to make the field optional.
+
+## Taproot Channel Proof and Verification
+
+Taproot channel funding transaction outputs will be SegWit V1 (P2TR) outputs of
+the following form:
+
+```
+OP_1 <taproot_output_key> 
+```
+
+where 
+```
+taproot_internal_key = MuSig2.KeyAgg(MuSig2.KeySort(bitcoin_key_1, bitcoin_key_2))
+
+taproot_output_key = taproot_internal_key + tagged_hash("TapTweak", taproot_internal_key) * G
+```
+
+`taproot_interal_key` is the aggregate key of `bitcoin_key_1` and
+`bitcoin_key_2` after first sorting the keys using the
+[MuSig2 KeySort][musig-keysort] algorithm and then running the
+[MuSig2 KeyAgg][musig-keyagg] algorithm.
+
+Then, to commit to spending the taproot output via the keyspend path, a
+[BIP86][bip86-tweak] tweak is added to the internal key to calculate the
+`taproot_output_key`.
+
+As with legacy channels, nodes will want to perform some checks before adding 
+an announced taproot channel to their routing graph:
+
+1. The funding output of the channel being advertised exists on-chain, is an
+   unspent UTXO and has a sufficient number of confirmations. To allow this 
+   check, the SCID of the channel will continue to be included in the channel
+   announcement.
+2. The funding transaction is sufficiently bound to the Lightning context. 
+   Nodes will do this by checking that they are able to derive the output key 
+   found on-chain by using the advertised `bitcoin_key_1` and `bitcoin_key_2` 
+   along with a [BIP86 tweak][bip86-tweak]. This provides a slightly weaker
+   binding to the LN context than legacy channels do but at least somewhat 
+   limits how the output can be spent due to the script-path being disabled.
+   The context binding with legacy channels is greater because an attacker 
+   trying to create gossip spam would need to create 2-of-2 multisig P2WSH 
+   outputs on-chain that would require them to then pay for the bytes of two
+   signatures at the time of spending the output. This extra cost provided some
+   extra deterrence for attackers. This deterrence is not present with P2TR 
+   transactions.
+3. The owners of `bitcoin_key_1` and `bitcoin_key_2` agree to be associated with  
```

Comment:
[From George](https://github.com/ellemouton/bolts/pull/2#discussion_r1135955383): 

Since the public announcements of channels pretty much reveal what the output is being used for, would it be fair to mention that this mainly benefits private channels?

Different phrasing: Taproot outputs all do look the same regardless of being used for a channel or not, but any participant in LN will eventually receive a (public) channel announcement that marks some taproot UTXO as a channel, so effectively it's only the private channels that can remain "incognito"